tests/llmcompressor/transformers/compression/test_decompress.py

```diff
@@ -0,0 +1,115 @@
+import copy
+import shutil
+import tempfile
+import unittest
+
+import torch
+from compressed_tensors import QUANTIZATION_CONFIG_NAME
+from compressed_tensors.compressors import ModelCompressor
+from compressed_tensors.quantization import QuantizationStatus
+from parameterized import parameterized_class
+from transformers import AutoConfig, AutoModelForCausalLM, AutoTokenizer
+
+from tests.testing_utils import parse_params, requires_gpu
+
+CONFIG_DIR = "tests/llmcompressor/transformers/compression/decompression_configs"
+
+
+@requires_gpu
+@parameterized_class(parse_params(CONFIG_DIR))
+class TestQuantizationMatches(unittest.TestCase):
+    """
+    Test decompression - given a skeleton model and path to the optimized model,
+    write the optimized model's safetensors to the skeleton model and decompress
+    Ex. write weight_scale to skeleton model and then fp4 -> fp16
+
+    """
+
+    compressed_model_stub = None
+    skeleton_model_stub = None
+
+    SAMPLE_INPUTS = [
+        "I love 4-bit quantization because",
+        "What is the capital of France?",
+        "def fibonacci(n):",
+    ]
+
+    @classmethod
+    def setUpClass(self):
+        self.test_dir = tempfile.mkdtemp()
+        self.tokenizer = AutoTokenizer.from_pretrained(self.compressed_model_stub)
+
+        self.compressed_model = AutoModelForCausalLM.from_pretrained(
+            self.compressed_model_stub,
+            torch_dtype="auto",
+            device_map="auto",
+        )
+
+        self.dense_model = AutoModelForCausalLM.from_pretrained(
+            self.skeleton_model_stub,
+            torch_dtype=self.compressed_model.dtype,
+            device_map=self.compressed_model.device,
+        )
+
+        assert not hasattr(
+            self.dense_model.model.layers[0].self_attn.q_proj, "weight_scale"
+        )
+
+        self.decompressed_model = None
+        config = AutoConfig.from_pretrained(self.compressed_model_stub)
+
+        compression_config = getattr(config, QUANTIZATION_CONFIG_NAME, None)
+        self.compressor = ModelCompressor.from_compression_config(compression_config)
+        self.compressor.quantization_config.quantization_status = (
+            QuantizationStatus.FROZEN
```

Comment:
instead of doing this manual decompression, we should be using the AutoModel pathway to decompress?

Reply:
Yeah thats more appropriate